src/lib/exeRunner.ts

```diff
@@ -26,7 +27,10 @@ export class ExeRunner {
             const stderr = new Array<string>();
 
             this.logger.info(`exe: ${this._exePath}, first arg of ${args.length}: ${args.length ? args[0]: '<none>'}`);
-            const process = spawn(this._exePath, args, { cwd: this.workingDir });
+            const process = spawn(this._exePath, args, {
+                cwd: this.workingDir,
+                env: Object.assign({ "PP_TOOLS_AUTOMATION_AGENT": getAutomationAgent.agent })
```

Comment:
I see that the cli-wrapper code uses Object.assign. In this file, the Object.assign doesn't need to be used.

In the cli-wrapper, we shouldn't use Object.assign either. Instead it should use the spread operator there to ensure it doesn't mutate the existing global state of the 'process'. That's actually a bug in that code.

Reply:
If you wanted this code to add on the new agent env string, this code should also use the spread operator:
```ts
import thisProcess from "process";

const process = spawn(this._exePath, args, {
   cwd: this.workingDir,
   env: { ...thisProcess.env,
      "PP_TOOLS_AUTOMATION_AGENT": getAutomationAgent.agent
   }
});
```

Reply:
@joem-msft see PR 25 in cli-wrapper.... w/o the explicit Object.assign(), the spawn cmd won't have the full set environment variables

Reply:
right. Let me clairify that there are two concerns here:
a) use TypeScript's spread operator syntax instead of using Object.assign.
b) the code in this PR doesn't use the current processor's env state (i.e. `process.env` is missing from its 2nd arg). So the call to Object.assign doesn't actually do anything.
c) we should make sure that these new properties override any that may already be set in the `process.env`, therefore the spread should come before the new env variables in the code.

The code above will give the correct and safe intended semantics.